pygmt/xarray/accessor.py

```diff
@@ -200,3 +231,29 @@ def gtype(self, value: GridType | int):
                 value, description="grid coordinate system type", choices=GridType
             )
         self._gtype = GridType(value)
+
+    @staticmethod
+    def _make_method(func):
+        """
+        Create a wrapper method for PyGMT grid-processing methods.
+
+        The :class:`xarray.DataArray` object is passed as the first argument.
+        """
+
+        @functools.wraps(func)
+        def wrapper(self, *args, **kwargs):
+            return func(self._obj, *args, **kwargs)
+
+        return wrapper
+
+    # Accessor methods for grid operations.
+    clip = _make_method(grdclip)
+    cut = _make_method(grdcut)
+    dimfilter = _make_method(dimfilter)
+    equalize_hist = _make_method(grdhisteq.equalize_grid)
```

Comment:
For consistency, I would prefer matching the original GMT `grdhisteq` name. But no strong opinion on this.

```suggestion
    histeq = _make_method(grdhisteq.equalize_grid)
```